src/Products/PluginIndexes/DateRangeIndex/DateRangeIndex.py

```diff
@@ -288,7 +275,7 @@ def _apply_index(self, request, resultset=None):
             until = multiunion(self._until.values(term))
 
             # Total result is bound by resultset
-            if REQUEST is None:
+            if cache is None:
                 until = intersection(resultset, until)
```

Comment:
This is not pull request related, but while reviewing this I stumbled over this:
We are in a block where it is guaranteed that ``resultset is None`` - why do we do ``intersection(None, until)``?